functions.php

```diff
@@ -113,9 +113,33 @@ function gesso_scripts() {
 	wp_register_script( 'gessoscripts', get_template_directory_uri() . '/js/dist/scripts.min.js', array( 'jquery', 'gessocommon', 'gessomodernizr' ), filemtime( get_template_directory() . '/js/dist/scripts.min.js' ) );
 	wp_enqueue_script( 'gessoscripts' );
 
-	// Add Google Fonts.
-	wp_register_style( 'gessoFonts', 'https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,300i,400,400i,600,600i,700,700i&display=swap', array(), $gesso_version );
-	wp_enqueue_style( 'gessoFonts' );
+	/**
+	* https://developer.wordpress.org/reference/hooks/style_loader_tag/
+	* Using style_loader_tag:
+	* - Enqueue the preconnect URL as a style
+	* - Change the rel attribute
+	* - Remove the type & media attributes
+	*/
+
+	// Enqueue Google Font.
+	wp_enqueue_style( 'google-fonts-preconnect', 'https://fonts.gstatic.com', false );
+	wp_enqueue_style( 'google-fonts', 'https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,300i,400,400i,600,600i,700,700i&display=swap', array( 'google-fonts-preconnect' ), null );
+
+	/**
+	 * Filter enqueue styles.
+	 */
+	function gesso_google_font_enqueued_styles( $html, $handle ) {
```

Comment:
@liquilife FYI, that this function is not setup with the correct number of arguments as it should be taking in 4(https://developer.wordpress.org/reference/hooks/style_loader_tag/). Please add the 2 missing arguments based on the WordPress documentation. Additionally, please add `@param` comments for all the function arguments, and you can literally copy the definitions from the WordPress documentation(https://developer.wordpress.org/reference/hooks/style_loader_tag/). FYI @coreylafferty

Reply:
@liquilife you can see more clearly the example comments from WP Core here: https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/class.wp-styles.php#L246